### ISSUE TITLE
[FEAURE REQUEST]: See the content of cookies

### ISSUE THREAD

**Issue by derei** [mrdokenny/Cookie-AutoDelete#800]
**Is your feature request related to a problem? Please describe.**

choosing which cookies to keep or discard from a list is... okay. Except that each website has its own cookies named the way the site wants... and users have to guess, or use another extension/application to figure stuff out.

**Describe the solution you'd like**

It would be lovely if (where possible), we could see some details about a selected cookie (eg: the content of it).

Thanks.

**Comment by kennethtran93** [comment]
What kind of details did you want to see in regards to the cookie?  At most it would be an additional cookie value field somewhere next to the cookie name as that's what webpages see - a list of key-value pairs (i.e. name=value; name2=value2; ).  Webpages can set additional options but cannot view them.

Normally it is best to use another extension for this.  In Firefox, this is a good one for searching up cookies and it's full contents that browsers (not webpages/sites) see:  https://addons.mozilla.org/en-US/firefox/addon/a-cookie-manager/  In Chrome...you'll have to search webextensions to find the one that suits you yourself.

Alternatively if you just want cookie names as per what we currently have, just make sure the expression domain is a specific hostname without the `*.` part in front.  Just like you said, the user has to guess regardless, especially if the cookie name and/or value is scrambled/random/encrypted and only the webpage will know how to handle it.

In essence this is a similar request to #756 and #272.

**Comment by derei** [comment]
i am mainly interested to save cookies that manage logins, per-site preferences (eg. currency for shopping websites, or cookie policies), shopping basket (although lately this is being managed on account level, server side, from what I could see)... and just guessing by name isn't always accurate. Being able to identify the content of such cookies may be helpful.

**Comment by kennethtran93** [comment]
I still think using a separate extension is better as it is usually made for such purposes.

The content of such cookies in this instance may just be cookie name and value.    Due to the limited spacing I think the design would have to be a readonly textarea in the expression option just below each cookie name 

Sample code may be: 
```html 
  <textarea name="message" rows="1" cols="15" wrap="off" spellcheck="false" autocomplete="off" readonly>TheCookieValueCanBeAVeryLongString</textarea>
```

which may look like this screenshot
![Screenshot_20200709_174149](https://user-images.githubusercontent.com/6724477/87103998-7dd8a400-c20b-11ea-8e07-fd58f3e52c53.png)

Keep in mind that the value is not necessarily `true` or `false`.  It can be encoded, encrypted, etc.  In addition, they may store everything into one cookie for parsing (e.g. preferences) like `samplePref=abc.1.def.0.ghi.xyzna.next.value` (e.g. key abc with value of 1, etc) or combine multiple other things under one cookie name.

**Comment by kennethtran93** [comment]
Also another thing to note is that if I were to implement cookie values and the cookie values are updated while the popup / Settings page with list of expressions is active/loaded, it will **NOT** show with the latest values.

**Comment by ghost** [comment]
Together with _Cookie Autodelete_ I use at this the Firefox _Cookie Quick Manager_ extension, to view cookies (which can be done within Firefox's Options page) but mainly to view their content (a feature Firefox doesn't provide anymore when it did previously to the Quantum revolution). Having _Cookie Autodelete_ provide this would be appreciated of course even if it's basically out of its scope. _Cookie Quick Manager_ provides as well cookie edition but that's not the point and would be definitely out of scope, just viewing the content would be a plus though.

**Comment by atomGit** [comment]
i disagree that this should be added - the info requested by the OP is already available in developer tools, at least for Firefox and i assume that "other" browser as well

**Comment by kennethtran93** [comment]
@atomGit While I do agree that the F12 / Developer Tools > Storage tab would provide the ability to view cookies, the only items that I don't see there, at least for Firefox, would be the firstPartyDomain for those that have FirstPartyIsolation turned on, as well as the cookie store / container ID, though that in itself may be useless as they should know which container they are in as they are visiting it actively.

For now I'm leaving it open to see if users do want it in CAD, but my stance is to leave it to other web extensions that allow view/edit of cookies in detail.

**Comment by ryan-carpenter** [comment]
Like @derei , I want to delete as many cookies as I can without causing myself too much inconvenience. I have tried using other cookie managers to inspect cookie contents, but this is less convenient that just letting Cookie Autodelete do its thing and then logging in again later when I need to.
I have a few sites in my whitelist, but I hesitate to use it more often, because I don't know which additional cookies I might save along with those that I want to keep. Cookie Autodelete shows me how many cookies are being saved or deleted, which is great. It also has a side effect of suggesting that whitelisting could undo the benefit of the extension (to a degree). My non-web-developer thinking tells me "Github wants to keep 10 cookies on my computer. I don't know what they are or what they are doing, but that seems like a lot". I don't know if it is technically feasible to identify the "login" cookies, but this is what I would like the grey list to do; not keep everything for a short time, but keep the "important" cookies for a longer time.

**Comment by zero77** [comment]
Don't know if it helps in any way but, this extension may be what you are after.
https://github.com/ysard/cookie-quick-manager
